src/common/settings.json

```diff
@@ -0,0 +1,731 @@
+[
+    {
+        "name": "access_mode",
+        "description": "Access mode of the database (AUTOMATIC, READ_ONLY or READ_WRITE)",
+        "type": "AccessMode",
+        "sql_type": "VARCHAR",
+        "scope": "global",
+        "custom_conversion_and_validation": true
+    },
+    {
+        "name": "allocator_background_threads",
+        "description": "Whether to enable the allocator background thread.",
+        "type": "bool",
+        "sql_type": "BOOLEAN",
+        "scope": "global",
+        "add_verification_in_both_SetReset": true
+    },
+    {
+        "name": "allocator_bulk_deallocation_flush_threshold",
+        "description": "If a bulk deallocation larger than this occurs, flush outstanding allocations.",
+        "type": "string",
+        "sql_type": "VARCHAR",
+        "scope": "global",
+        "custom_conversion_and_validation": true
```

Comment:
Unfortunately, that's not possible because there are a couple of settings in which Set/Reset's functionality differs a lot from just setting one variable.

For example:
https://github.com/duckdb/duckdb/blob/af39bd0dcf66876e09ac2a7c3baa28fe1b301151/src/main/settings/settings.cpp#L380
https://github.com/duckdb/duckdb/blob/af39bd0dcf66876e09ac2a7c3baa28fe1b301151/src/main/settings/settings.cpp#L396
https://github.com/duckdb/duckdb/blob/af39bd0dcf66876e09ac2a7c3baa28fe1b301151/src/main/settings/settings.cpp#L677

Reply:
Okay fair enough! lets go with `custom_implementation` instead of `custom_conversion_and_validation`